client/client.go

```diff
@@ -200,7 +208,7 @@ func (client *Client) writeLoop() {
 	// Pipeline requests; but only write them one at a time. To allow multiple
 	// goroutines to all write as quickly as possible, uses a channel and the
 	// writeLoop lives in a separate goroutine.
-	for req := range client.outbound {
+	for req := range client.loadChans().outbound {
```

Comment:
Worth noting that the expression in [range is not recalculated](https://play.golang.org/p/_P33M1fLCIX)

Reply:
yeah we don't want it to recalculate, we want it to bounce out of this loop when it's done